syncer/syncer.go

```diff
@@ -1404,16 +1446,47 @@ func (s *Syncer) syncDML(
 
 // Run starts running for sync, we should guarantee it can rerun when paused.
 func (s *Syncer) Run(ctx context.Context) (err error) {
-	tctx := s.tctx.WithContext(ctx)
+	runCtx, runCancel := context.WithCancel(context.Background())
+	defer runCancel()
+	tctx := s.tctx.WithContext(runCtx)
 
 	defer func() {
 		if s.done != nil {
 			close(s.done)
 		}
 	}()
 
+	go func() {
+		<-ctx.Done()
+		select {
+		case <-runCtx.Done():
+		default:
+			tctx.L().Info("received subtask's done")
+
+			s.waitTransactionLock.Lock()
+			if s.isTransactionEnd {
+				s.jobWg.Wait()
```

Comment:
If we wait here, it seems that we can't add dml job anymore. Will this cause a problem?

Reply:
It just waits added job completed.